app/screens/find_channels/filtered_list/filtered_list.tsx

```diff
@@ -1,44 +1,176 @@
     flex: {flex: 1},
+    loading: {
+        height: 32,
+        width: 32,
+        justifyContent: 'center',
+    },
     noResultContainer: {
         flexGrow: 1,
         alignItems: 'center',
         justifyContent: 'center',
     },
 });
 
-const UnfilteredList = ({close, keyboardHeight, showTeamName, term}: Props) => {
+export const MAX_RESULTS = 20;
+
+const sortyByUserOrChannel = <T extends Channel |UserModel>(locale: string, teammateDisplayNameSetting: string, a: T, b: T): number => {
+    const aDisplayName = 'display_name' in a ? a.display_name : displayUsername(a, locale, teammateDisplayNameSetting);
+    const bDisplayName = 'display_name' in b ? b.display_name : displayUsername(b, locale, teammateDisplayNameSetting);
+
+    return aDisplayName.toLowerCase().localeCompare(bDisplayName.toLowerCase(), locale, {numeric: true});
+};
+
+const FilteredList = ({
+    archivedChannels, close, channelsMatch, channelsMatchStart, currentTeamId,
+    keyboardHeight, loading, onLoading, restrictDirectMessage, showTeamName,
+    teamIds, teammateDisplayNameSetting, term, usersMatch, usersMatchStart,
+}: Props) => {
+    const bounce = useRef<DebouncedFunc<() => void>>();
+    const mounted = useRef(false);
     const serverUrl = useServerUrl();
-    const [data] = useState([]);
+    const theme = useTheme();
+    const {locale, formatMessage} = useIntl();
     const flatListStyle = useMemo(() => ({flexGrow: 1, paddingBottom: keyboardHeight}), [keyboardHeight]);
+    const [remoteChannels, setRemoteChannels] = useState<RemoteChannels>({archived: [], startWith: [], matches: []});
+    const totalLocalResults = channelsMatchStart.length + channelsMatch.length + usersMatchStart.length;
+
+    const search = async () => {
+        onLoading(true);
+        if (mounted.current) {
+            setRemoteChannels({archived: [], startWith: [], matches: []});
+        }
+        const lowerCasedTerm = (term.startsWith('@') ? term.substring(1) : term).toLowerCase();
+        if ((channelsMatchStart.length + channelsMatch.length) < MAX_RESULTS) {
+            if (restrictDirectMessage) {
+                searchProfiles(serverUrl, lowerCasedTerm, {team_id: currentTeamId, allow_inactive: true});
+            } else {
+                searchProfiles(serverUrl, lowerCasedTerm, {allow_inactive: true});
+            }
+        }
+
+        if (!term.startsWith('@')) {
+            if (totalLocalResults < MAX_RESULTS) {
+                const {channels} = await searchAllChannels(serverUrl, lowerCasedTerm, true);
+                if (channels) {
+                    const existingChannelIds = new Set(channelsMatchStart.concat(channelsMatch).concat(archivedChannels).map((c) => c.id));
+                    const filteredChannels = channels.filter((c) => !existingChannelIds.has(c.id) && teamIds.has(c.team_id));
+                    const startWith = filteredChannels.filter((c) => c.display_name.toLowerCase().startsWith(term) && c.delete_at === 0).
```

Comment:
Just to doublecheck: the term has been already set to lowercase? Looking at line 92, that you are using a different variable for this purpose, I guess it is not.

If not, a term like "Hello" will never match anything.